compute_tools/src/bin/fast_import.rs

```diff
@@ -677,11 +678,35 @@ pub(crate) async fn main() -> anyhow::Result<()> {
                 spec,
                 source_connection_string,
                 destination_connection_string,
-                args.working_directory,
+                args.working_directory.clone(),
                 args.pg_bin_dir,
                 args.pg_lib_dir,
             )
-            .await?;
+            .await
+        }
+    };
+
+    if let Some(s3_prefix) = args.s3_prefix {
+        info!("write job status to s3");
+        {
+            let status_dir = args.working_directory.join("status");
+            if std::fs::exists(&status_dir)?.not() {
+                std::fs::create_dir(&status_dir).context("create status directory")?;
+            }
+            let status_file = status_dir.join("fast_import");
+            let res_obj = if res.is_ok() {
+                serde_json::json!({"done": true})
+            } else {
+                serde_json::json!({"done": false, "error": res.unwrap_err().to_string()})
+            };
```

Comment:
nit: you could match to avoid the unwrap:

```
let res_obj = match res {
  Ok(_) => serde_json::json!({"done": true}),
  Err(err) => serde_json::json!({"done": false, "error": err.to_string()})
};
```

Reply:
thank you, fixed!